crates/sui-core/src/authority.rs

```diff
@@ -360,29 +360,47 @@ impl AuthorityState {
         }
     }
 
+    pub async fn handle_node_sync_transaction(
+        &self,
+        certificate: CertifiedTransaction,
+        // Signed effects is signed by only one validator, it is not a
+        // CertifiedTransactionEffects. The caller of this (node_sync) must promise to
+        // wait until it has seen at least f+1 identifical effects digests matching this
+        // SignedTransactionEffects before calling this function, in order to prevent a
+        // byzantine validator from giving us incorrect effects.
+        signed_effects: SignedTransactionEffects,
```

Comment:
Do we not have a Certified effects struct yet?

Reply:
There is such a thing, and you can assemble it from a quorum of SignedTransactionEffects.

We don't need to do that here, because we verify the effects by getting at least f+1 identical digests, which tells us that we have the correct effects structure (all honest validators will produce the same effects, of f+1, 1 must be honest).

What a CertifiedTransactionEffects lets you do is prove to someone else that the tx is final. We only need to prove it to ourselves.